pkg/debuginfo/debuginfod.go

```diff
@@ -182,6 +184,10 @@ func (c *HTTPDebuginfodClient) request(ctx context.Context, u url.URL, buildID s
 	// this request will result in a binary object that contains the customary .*debug_* sections.
 	u.Path = path.Join(u.Path, "buildid", buildID, "debuginfo")
 
+	// Uses the default tracer provider and propagator that's set in tracer package,
+	// or from the span context passed in.
+	ctx = httptrace.WithClientTrace(ctx, otelhttptrace.NewClientTrace(ctx))
```

Comment:
This might be cleaner than the transport wrapper. This is what we used on the agent side. I wonder how this would interact with the transport wrapper.

Reply:
Oh nice I didn't see this. Should we maybe then use the [`WithTracerProvider`](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/httptrace/otelhttptrace#WithTracerProvider) so we don't use the global one?